lib/matplotlib/backend_bases.py

```diff
@@ -138,6 +145,35 @@ def get_registered_canvas_class(format):
     return backend_class
 
 
+class MainLoopBase(object):
```

Comment:
``abc`` should work here, but I don't see much point, we don't need ``end`` as an abstract method, for example in the qt backend I don't specify ``end``, as far as I could tell in my research ``Qt`` doesn't have a method to terminate the mainloop, qt internally handles the automatic release of the loop once all windows have been closed.  When developing it should become readily apparent if we need that method.

As for ``begin`` I wondered if we couldn't put ``do_nothing_show`` in there at a later date.

As far as I see MEP27 I see refactoring out ``Gcf`` as a nice side-effect of us refactoring the GUI backends, i.e. refactoring ``FigureManagerBackend`` into ``FigureManager`` ``WindowBackend`` and ``MainLoopBackend``.  Of the non GUI backneds only the ``Pdf`` backend uses ``Gcf`` and it uses it in a way we can't refactor, I think it needs a non BC change to rid it of Gcf, perhaps something for MPL 2.0...